comp/dogstatsd/server/component.go

```diff
@@ -0,0 +1,67 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package server implements a component to run the dogstatsd server
+package server
+
+import (
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/dogstatsd"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"go.uber.org/fx"
+)
+
+// team: Agent Metrics Logs
+
+// Component is the component type.
+type Component interface {
+
+	// Start starts the dogstatsd server
+	Start(demultiplexer aggregator.Demultiplexer) error
```

Comment:
For `Start` / `Stop` you should use [Lifecycle](https://github.com/DataDog/datadog-agent/blob/aeb058147baa26e9c7c4708ab46f16512dc34ad0/docs/components/fx.md#lifecycle)

Reply:
I am not seeing a way to make this work with `Start` taking an argument (demux). 

Until Demux is a component, we will have to provide it as an argument to `Start`. Once Dmux is a component and can provide it in the constructor. I think this will work. Because right now demux has to be created and configured before dogstatsd. 

Does this make sense or is there a better way to use lifecycles for this?

Reply:
Indeed, you have to inject `aggregator.Demultiplexer` in the constructor. Can you track this change?